tests/test_collection.py

```diff
@@ -38,3 +38,11 @@ def test_collection_not_published_404(client, draft_collection):
     url = reverse('product:collection', kwargs=url_kwargs)
     response = client.get(url)
     assert response.status_code == 404
+
+def test_collection_not_yet_published(admin_client, client, not_yet_available_collection):
```

Comment:
We could add a multiple cases test here (parametrized test) instead of having two tests (`test_collection_not_published_404`). Also, we could have a more descriptive test name, as the test above this one: `test_collection_not[_yet]_published_returns_404`.